src/utils/getSmartContract.tsx

```diff
@@ -12,42 +13,48 @@ const { notifyError, notifySuccess } = useNotification()
 export type AmountsOut = Array<BigNumber>
 
 export type TokenContracts = {
-    tokenAContract: Contract | null
-    tokenBContract: Contract | null
+  tokenAContract: Contract | null
+  tokenBContract: Contract | null
 }
 
 export const getTokenContracts = (
-    tokenAAddr: string,
-    tokenBAddr: string,
-    provider: Provider,
+  tokenAAddr: string,
+  tokenBAddr: string,
+  provider: Provider,
 ): TokenContracts | null => {
-    try {
-        const tokenAContract = new ethers.Contract(tokenAAddr, erc20ABI, provider)
-        const tokenBContract = new ethers.Contract(tokenBAddr, erc20ABI, provider)
-
-        return {
-            tokenAContract: tokenAContract,
-            tokenBContract: tokenBContract,
-        }
-    } catch (error) {
-        let message
-        if (error instanceof Error) message = error.message
-        else message = String(error)
+  try {
+    const tokenAContract = new ethers.Contract(tokenAAddr, erc20ABI, provider)
+    const tokenBContract = new ethers.Contract(tokenBAddr, erc20ABI, provider)
 
-        notifyError(message)
+    return {
+      tokenAContract: tokenAContract,
+      tokenBContract: tokenBContract,
     }
-    return null
+  } catch (error) {
+    let message
+    if (error instanceof Error) message = error.message
+    else message = String(error)
+
+    notifyError(message)
+  }
+  return null
 }
 
 export const getEstimatedOutput = async (
-    tokenAddrs: Array<string>,
-    inputAmt: string,
-    provider: Provider
+  tokenAddrs: Array<string>,
+  inputAmt: string,
+  provider: Provider
 ): Promise<AmountsOut | undefined> => {
-    const routerContract = new ethers.Contract(process.env.NEXT_PUBLIC_IUNISWAP_V2_ROUTER_02_ADDR as string, IUniswapV2Router02ABI, provider as Provider)
-    let convertedInput = ethers.utils.parseEther(inputAmt)
-    const amountsOut = await routerContract.getAmountsOut(convertedInput, [tokenAddrs[0], tokenAddrs[1]])
-    return amountsOut
+  const routerContract = new ethers.Contract(process.env.NEXT_PUBLIC_IUNISWAP_V2_ROUTER_02_ADDR as string, IUniswapV2Router02ABI, provider as Provider)
+  let convertedInput = ethers.utils.parseEther(inputAmt)
+  const amountsOut = await routerContract.getAmountsOut(convertedInput, [tokenAddrs[0], tokenAddrs[1]])
+  return amountsOut
+}
+
+export const calcPoolKey = (cfmm: string): string => {
```

Comment:
should pass protocol number as a parameter, not have it hardcoded to 1 here

Reply:
done